### PR TITLE
Add feature to inverse read/write access logic

### PR DESCRIPTION
**Pull request by ggrebert**
@cowtowncoder replacement of #2966 

fix https://github.com/FasterXML/jackson-databind/issues/2951

### CODE REVIEW

**Comment by JooHyukKim** [comment]
Did u mean to PR against master(version3) branch?
Asking this since it seems like something that can be applied to 2.19

**Comment by ggrebert** [comment]
the target branch of this PR is `2.19`

File: src/main/java/com/fasterxml/jackson/databind/MapperFeature.java [hunk 1]
```diff
@@ -280,6 +281,14 @@ public enum MapperFeature implements ConfigFeature
      */
     OVERRIDE_PUBLIC_ACCESS_MODIFIERS(true),
 
+    /**
+     * Feature that inverse logic in {@link JsonProperty#access}
+     * for <code>READ_ONLY</code> and <code>WRITE_ONLY</code>.
+     *<p>
+     * Feature is disabled by default.
+     */
```
Comment (JooHyukKim) [inline src/main/java/com/fasterxml/jackson/databind/MapperFeature.java:289]:
```suggestion
     * 
     * @since 2.19
     */
```

**Review by JooHyukKim — COMMENTED** [review]
Looks clean

**Comment by cowtowncoder** [comment]
Looks clean! One quick question @ggrebert: have we asked for (and received) a CLA earlier?
It's from

    https://github.com/FasterXML/jackson/blob/master/contributor-agreement.pdf .

If not, we'd need it before merging (just once; good for any and all future contributions).
The usual way is to print it, fill & sign, scan/photo, email to `cla` at fasterxml dot com.
Once I get it I can merge the PR to get feature added in for 2.19!

Thank you again for contributing this feature; looking forward to merging it.

**Comment by ggrebert** [comment]
CA signed and sended

**Comment by cowtowncoder** [comment]
@ggrebert Almost there -- I asked for 2 minor changes for CLA and then we should be good. Apologies for the hassle, looking forward to getting this merged!

**Comment by ggrebert** [comment]
> @ggrebert Almost there -- I asked for 2 minor changes for CLA and then we should be good. Apologies for the hassle, looking forward to getting this merged!

done

**Comment by cowtowncoder** [comment]
CLA good!

File: src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertyBuilder.java [hunk 2]
```diff
@@ -946,6 +946,20 @@ public JsonProperty.Access removeNonVisible(boolean inferMutators,
         if (acc == null) {
             acc = JsonProperty.Access.AUTO;
         }
+
+        // [databind#2951] add feature to inverse access logic
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertyBuilder.java:950]: I think this should actually be moved to occur within "findAccess()" -- I can change that

**Review by cowtowncoder — APPROVED** [review]

**Comment by cowtowncoder** [comment]
Merged in 2.19 for 2.19.0. Thank you again @ggrebert !

**Pull request merged**